Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/xml/XMLConceptMessage.java

```diff
@@ -73,10 +73,19 @@ public String getEntry(ConceptEntry entry, ConceptType type) {
         List<ChangeEvent> changeEvents = entry.getChangeEvents();
         Collections.sort(changeEvents);
 
-        sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
-        sb.append(StringEscapeUtils.escapeXml10(
-                changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
-        sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+        if (changeEvents != null && !changeEvents.isEmpty()) {
+            sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+            sb.append(StringEscapeUtils.escapeXml10(
+                    changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
+            sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+        }
+
+        else {
+            // creator id
+            sb.append("<" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+            sb.append(StringEscapeUtils.escapeXml10(entry.getCreatorId() != null ? entry.getCreatorId().trim() : ""));
+            sb.append("</" + XMLConstants.NAMESPACE_PREFIX + ":" + XMLConstants.CREATOR_ID + ">");
+        }
```

Comment:
There was an issue while loading the creator id. Creator id needs to be retrieved from changevent only when creator id is in changeevent, else creator id needs to be fetched  as in the old design.